File: BE/src/main/java/com/self/roomescape/entity/Theme.java
package com.self.roomescape.entity;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;

@Entity
@Getter
@Setter
@ToString
@Table(name = "theme")
public class Theme {

    @Id
    @Column(name = "tid")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int tid;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cid")
    @ToString.Exclude
    private Cafe cafe;
    private int difficulty;
    private String tname;
    private String img;
    private String description;
    private String rplayer;
    private String time;
    private String genre;
    private String type;
    private float grade;
    private int reviewCnt;
    private String activity;

    @Transient
    private int count;

}

File: AOS/app/src/main/java/com/ssafy/moabang/data/model/dto/Theme.kt
package com.ssafy.moabang.data.model.dto

import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.parcelize.Parcelize

@Entity
@Parcelize
data class Theme(
    val time: String,
    val type: String,
    val description: String?,
    @PrimaryKey val tid: Int,
    val difficulty: Int,
    val tname: String,
    val rplayer: String,
    val genre: String,
    val grade: Double,
    val activity: String,
    val cid: Int,
    var cname: String,
    val img: String,
    var island: String, // var -> val
    var si: String, // var -> val
    var url: String,
    var islike: Boolean,
    val min_player: Int,
    val max_player: Int,
    var count : Int,
): Parcelable